Build fails on fresh clone - shell-core dependency not built automatically

## Problem

Running `npm run build` from `lua-learning-website` fails on a fresh clone with TypeScript errors:

```
error TS2307: Cannot find module '@lua-learning/shell-core' or its corresponding type declarations.
```

This happens because:
1. `lua-learning-website` depends on `@lua-learning/shell-core` (a local workspace package)
2. The `shell-core` package's `dist/` directory is not committed (correctly in `.gitignore`)
3. **CI workflows correctly build `shell-core` before the website** (no issue there)
4. **Local build script (`npm run build`) doesn't build `shell-core` first**
5. **`docs/getting-started.md` doesn't mention the shell-core dependency**

## Impact

- Any developer cloning the repo for the first time cannot build
- Manual workaround: `npm run build:shell-core` from root, then build website

## Proposed Solution

1. **Add `prebuild` script to `lua-learning-website/package.json`:**
   ```json
   "prebuild": "npm run build -w @lua-learning/shell-core"
   ```
   This ensures shell-core is always built before the website build.

2. **Update `docs/getting-started.md`** to document the workspace structure and dependency.

3. **Consider adding a root-level build script** that builds all packages in dependency order.

## Tasks

- [ ] Add `prebuild` script to `lua-learning-website/package.json`
- [ ] Update `docs/getting-started.md` to mention shell-core package
- [ ] Add root-level `build` script that builds all packages
- [ ] Test fresh clone workflow

## Labels

bug, developer-experience

--- a/docs/getting-started.md
+++ b/docs/getting-started.md
@@ -8,6 +8,20 @@ This guide will help you set up your development environment for LuaInTheWeb.
 - npm 9+ or yarn
 - Git
 
+## Project Structure
+
+LuaInTheWeb uses npm workspaces to manage multiple packages:
+
+```
+LuaInTheWeb/
+├── lua-learning-website/    # Main React application
+├── packages/
+│   └── shell-core/          # @lua-learning/shell-core - Shell emulation library
+└── package.json             # Root workspace configuration
+```
+
+The `lua-learning-website` depends on `@lua-learning/shell-core`, which provides shell emulation functionality. This package is built automatically when you run `npm run build` in the website directory.
+
 ## Setup
 
 1. Clone the repository:
@@ -16,18 +30,36 @@ This guide will help you set up your development environment for LuaInTheWeb.
    cd LuaInTheWeb
    ```
 
-2. Install dependencies:
+2. Install dependencies (from the root directory):
    ```bash
-   cd lua-learning-website
    npm install
    ```
+   This installs dependencies for all workspace packages.
 
 3. Start the development server:
    ```bash
+   cd lua-learning-website
    npm run dev
    ```
 
 4. Open http://localhost:5173 in your browser
+
+## Building
+
+To build for production:
+
+```bash
+cd lua-learning-website
+npm run build
+```
+
+This automatically builds the `@lua-learning/shell-core` dependency first via the `prebuild` script, then builds the website.
+
+Alternatively, build all packages from the root:
+
+```bash
+npm run build
+```
 
 ## Available Scripts
 

--- a/lua-learning-website/package.json
+++ b/lua-learning-website/package.json
@@ -5,6 +5,7 @@
   "type": "module",
   "scripts": {
     "dev": "vite",
+    "prebuild": "npm run build -w @lua-learning/shell-core",
     "build": "tsc -b && vite build",
     "lint": "eslint .",
     "preview": "vite preview",
@@ -49,7 +50,7 @@
     "eslint-plugin-react-hooks": "^5.2.0",
     "eslint-plugin-react-refresh": "^0.4.22",
     "globals": "^16.4.0",
-    "jsdom": "^27.2.0",
+    "jsdom": "^27.3.0",
     "typescript": "~5.9.3",
     "typescript-eslint": "^8.45.0",
     "vite": "^7.1.7",

--- a/package.json
+++ b/package.json
@@ -6,6 +6,7 @@
     "lua-learning-website"
   ],
   "scripts": {
+    "build": "npm run build -w @lua-learning/shell-core && npm run build -w lua-learning-website",
     "build:shell-core": "npm run build -w @lua-learning/shell-core",
     "test:shell-core": "npm run test -w @lua-learning/shell-core"
   }

